portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,26 +33,46 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> sampleMessages = new ArrayList<>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
-    String messages = convertToJsonString(sampleMessages);
-    response.setContentType("text/html;");
-    response.getWriter().println(messages);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
nit: since this var is not used again, you can directly chain this call with the prepare call below.

Reply:
go it